sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -574,6 +574,21 @@ case class DeleteFromTable(
   override def children: Seq[LogicalPlan] = child :: Nil
 }
 
+/**
+ * Update the data of table that specified with the condition with the given updated values.
+ * NOTE: Considering nested fields, we let the type of `attrs` to be `Seq[Expression]` becuase
+ * nested fields may be resolved to objects of type like `GetStructField`. However, currently
+ * we only support top-level fields, in which case `attrs` is actually of type `Seq[Attribute]`.
+ */
+case class UpdateTable(
+    child: LogicalPlan,
```

Comment:
Why not use `NamedRelation`?

Reply:
Sorry @rdblue , using `LogicalPlan` is an inertia and I didn't consider it seriously before. `NamedRelation` seems a better choice.